runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -159,8 +159,12 @@ private Layout getLayoutUninterruptibly() {
 
             // Invoking the systemDownHandler if the client cannot connect to the server.
             if (--systemDownTriggerCounter <= 0) {
-                runtime.systemDownHandler.run();
+                log.info("layoutHelper: Invoking the systemDownHandler.");
+                runtime.getParameters().getSystemDownHandler().run();
             }
+            // Minimum counter value should be 0.
+            systemDownTriggerCounter = Math.max(0, systemDownTriggerCounter);
```

Comment:
if we are going to do it this way, then why not do this way
if(Math.max(0, --systemDownTriggerCounter) == 0) {

}

Reply:
We will not update the counter with the value then. This causes us to log negative numbers in the log statements.

Reply:
yes or you can do a better job by updating the number of attempts you have made so far.

Reply:
Done.